### PR TITLE
Support JuMP SparseAxis

### PR DESCRIPTION
**Pull request by guimarqu**
Implementing Benoit suggestion.

### CODE REVIEW

File: src/axis.jl [hunk 1]
```diff
@@ -2,10 +2,47 @@ import Base.length
...
+# Permit the access to the entry of an array using an AxisId.
+Base.to_index(i::AxisId) = i.indice
+
+# Allow matching of AxisId key using the value of field indice (dict.jl:289)
+# and vice-versa
+Base.isequal(i::T, j::AxisId{N,T}) where {N,T} = isequal(i, j.indice)
+Base.isequal(i::AxisId{N,T}, j::T) where {N,T} = isequal(i.indice, j)
+
+iterate(i::AxisId) = (i, nothing)
+iterate(i::AxisId, ::Any) = nothing
```
Comment (vitornesello) [inline src/axis.jl]: This means you cannot have a vector of `AxisId`?

File: src/decomposition.jl [hunk 2]
```diff
@@ -59,22 +57,31 @@ function look_for_dec_axis(container::JuMP.Containers.DenseAxisArray)::Vector{Ax
...
 end
 
-look_for_dec_axis(constr::JuMP.ConstraintRef) = Vector{Axis}()
-look_for_dec_axis(var::JuMP.VariableRef) = Vector{Axis}()
-look_for_dec_axis(vars::Array{<:JuMP.VariableRef, N}) where N =  Vector{Axis}()
-look_for_dec_axis(constrs::Array{<:JuMP.ConstraintRef, N}) where N =  Vector{Axis}()
+function look_for_dec_axis(tree::Tree, container::JC.SparseAxisArray)::Vector{Axis}
+    dec_axes = Vector{Axis}()
+    container_keys = collect(keys(container.data))
+    for indice in container_keys[1]
```
Comment (vitornesello) [inline src/decomposition.jl]: It is a bit obscure how and why it works..

Comment (vitornesello) [reply src/decomposition.jl]: Maybe a comment would help here

File: src/decomposition.jl [hunk 3]
```diff
@@ -85,10 +92,36 @@ function get_indices_of_obj_in_partition(
...
+            end
+        end
+    end
+    return indices
+end
+
+get_indices_of_obj_in_partition(obj_ref::JuMP.ConstraintRef, _) = ()
+get_indices_of_obj_in_partition(obj_ref::JuMP.VariableRef, _) = ()
+get_indices_of_obj_in_partition(obj_ref::Array{<:JuMP.VariableRef, N}, _) where N = ntuple(i -> Colon(), N)
+get_indices_of_obj_in_partition(obj_ref::Array{<:JuMP.ConstraintRef, N}, _) where N = ntuple(i -> Colon(), N)
```
Comment (vitornesello) [inline src/decomposition.jl]: I don't understand how this works

File: src/decomposition.jl [hunk 4]
```diff
@@ -85,10 +92,36 @@ function get_indices_of_obj_in_partition(
     return tuple
 end
 
-get_indices_of_obj_in_partition(obj_ref::JuMP.ConstraintRef, _, _) = ()
-get_indices_of_obj_in_partition(obj_ref::JuMP.VariableRef, _, _) = ()
-get_indices_of_obj_in_partition(obj_ref::Array{<:JuMP.VariableRef, N}, _, _) where N = ntuple(i -> Colon(), N)
-get_indices_of_obj_in_partition(obj_ref::Array{<:JuMP.ConstraintRef, N}, _, _) where N = ntuple(i -> Colon(), N)
+function get_indices_of_obj_in_partition(
```
Comment (vitornesello) [inline src/decomposition.jl]: this is also a bit obscure. Can you add a comment explaining what it does?
I searches all the indices in `obj_ref` that corresponds to `dec_axes_val` in some way? Also, what is dec_axes_val? you type these functions to improve readability?

File: src/axis.jl (hunk 1 above)
Comment (guimarqu) [reply src/axis.jl]: It means that you can iterate over an `AxisId` but its a singleton containing only the `AxisId`.
```julia
for i in axisid
 # i is the axisid
end
```
It's like Number.

**Pull request merged**